src/kernels/jupyter/session/jupyterSession.node.ts

```diff
@@ -204,7 +206,12 @@ export class JupyterSession extends BaseJupyterSession {
         }
     }
 
-    private async createBackingFile(): Promise<Contents.IModel | undefined> {
+    private async createBackingFile(): Promise<string | undefined> {
+        if (this.connInfo.localLaunch) {
```

Comment:
Found that we don't need a backing file at all, we can start a session without a valid file.
I think creating this backing file and associating it with the session results in checkpoints getting created.
If it still gets created, it will get created in the temporary files directory.

The change ensures we use temporary file for the backing file.

Reply:
this change fixes the issue repirted (confirmed by user here)
https://github.com/microsoft/vscode-jupyter/issues/6510#issuecomment-1086038321

Reply:
So we do need a backing file? But you're putting it into the temp folder?

Reply:
based on the comments of the jupyter code, its not required, however, we use the value in the session to display the path of the notebook in the kernel picker drop down. so we do need a filepath (we need something).

also, i feel it's safe to provide a valid file path (that doesn't exist, but can be created) rather than failing somewhere because such a file cannot be created. ie this way its a simple work around, provide a temp file (if jupyter needs it, they'll create the file)

Reply:
> putting it into the temp folder?

yes, this way if checkpoints get created, it will be created in that temp directory, and users will never see them. ie it's not in the current workspace folder